hamilton/experimental/h_async.py

```diff
@@ -0,0 +1,151 @@
+
+        @param node: Node to wrap
+        @param kwargs: Keyword arguments (either coroutines or raw values) to call it with
+        @return: A coroutine
+        """
+        callabl = node.callable
+
+        async def new_fn(fn=callabl, **fn_kwargs):
+            fn_kwargs = await await_dict_of_tasks({key: asyncio.create_task(process_value(value)) for key, value in fn_kwargs.items()})
+            if inspect.iscoroutinefunction(fn):
+                return await(fn(**fn_kwargs))
+            return fn(**fn_kwargs)
+
+        coroutine = new_fn(**kwargs)
+        task = asyncio.create_task(coroutine)
+        return task
+
+    def build_result(self, **outputs: typing.Dict[str, typing.Any]) -> typing.Any:
+        """Currently this is a no-op -- it just delegates to the resultsbuilder.
+        That said, we *could* make it async, but it feels wrong -- this will just be
+        called after `raw_execute`.
+
+        @param outputs: Outputs (awaited) from the graph.
+        @return: The final results.
+        """
+        return self.result_builder.build_result(**outputs)
+
+
+class AsyncDriver(driver.Driver):
+    def __init__(self, config, *modules, result_builder: Optional[base.ResultMixin] = None):
+        """Instantiates an asynchronous driver.
+
+        @param config: Config to build the graph
+        @param modules: Modules to crawl for fns/graph nodes
+        @param result_builder: Results mixin to compile the graph's final results. TBD whether this should be included in the long run.
+        """
+        super(AsyncDriver, self).__init__(config, *modules, adapter=AsyncGraphAdapter(result_builder=result_builder))
+
+    async def raw_execute(self,
+                          final_vars: typing.List[str],
+                          overrides: Dict[str, Any] = None,
+                          display_graph: bool = False,  # don't care
+                          inputs: Dict[str, Any] = None) -> Dict[str, Any]:
+        """Executes the graph, returning a dictionary of strings (node keys) to final results.
+
+        Note inputs can be coroutines as well if you desire.
+
+        @param final_vars: Variables to execute (+ upstream)
+        @param overrides: Overrides for nodes
+        @param display_graph: whether or not to display graph -- this is not supported.
+        @param inputs:  Inputs for DAG runtime calculation
+        @return: A dict of key -> result
+        """
+        nodes, user_nodes = self.graph.get_upstream_nodes(final_vars, inputs)
+        memoized_computation = dict()  # memoized storage
+        self.graph.execute(nodes, memoized_computation, overrides, inputs)
+        if display_graph:
+            raise ValueError(f'display_graph=True is not supported for the async graph adapter. '
+                             f'Instead you should be using visualize_execution.')
+        return await await_dict_of_tasks({key: asyncio.create_task(process_value(memoized_computation[key])) for key in final_vars})
```

Comment:
```suggestion
        task_dict = {key: asyncio.create_task(process_value(memoized_computation[key])) for key in final_vars}
        return await await_dict_of_tasks(task_dict)
```
Also is `process_value` needed here? Seems a bit redundant with the result being wrapped in a task and that awaited?

Reply:
`process_value` is needed but `create_task` is not I think...

Reply:
okay. But this seems like an important line that should be at least two... break out the dict comprehension at least.